### DISCUSSION TITLE
How to filter a row depending on which parquet file it is in?

### DISCUSSION THREAD

**Discussion by SteveLauC** [apache/datafusion#7979] [Q&A] [answered]
Say I have multiple parquet files (`0.parquet`, `1.parquet`), when running queries against them, I want to filter specific rows depending on the source file by tweaking the generated physical plan, for example, rows that come from `0.parquet` and have `ID`(assume `ID` is a field stored in the parquet file) 0 should be filtered out, rows that are from `1.parquet` and have `ID` 1 should be filtered out...

I see that we can use [FilterExec](https://docs.rs/datafusion/latest/datafusion/physical_plan/filter/struct.FilterExec.html) to do the filter functionality, but with [`PhysicalPlan`](https://docs.rs/datafusion/latest/datafusion/physical_plan/trait.PhysicalExpr.html), it seems that we ONLY know the row itself (`RecordBatch`), not the parquet file information:

```
fn evaluate(
    &self,
    batch: &RecordBatch
) -> Result<ColumnarValue, DataFusionError>;
```

**Comment by Jefffrey** [comment, marked as answer]
I'm guessing something like this is probably what you're looking for: https://github.com/apache/arrow-datafusion/issues/6051

**Reply by SteveLauC** [reply]
Thanks for showing me this, yeah, I believe this will help, but I am more expecting it to be an execution context rather than a built-in function though
